### PR TITLE
Add Virginia Adjusted Gross Income

### PR DESCRIPTION
**Pull request by Kiki-67**
Add Virginia Adjusted Gross Income

### Summary
:sparkles::memo::us:

This pull request adds a new variable `va_agi` for the Virginia Adjusted Gross Income (VAGI) and the corresponding additions and subtractions that are used to calculate it from the Federal Adjusted Gross Income (FAGI). It also updates the changelog entry to reflect the new feature.

> _New variable_
> _`va_agi` joins the model_
> _Autumn tax season_

### Walkthrough
*  Bump minor version and add VAGI variable to changelog ([link](https://github.com/PolicyEngine/policyengine-us/pull/3341/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))
*  Define VAGI as the sum of FAGI and state-specific additions and subtractions ([link](https://github.com/PolicyEngine/policyengine-us/pull/3341/files?diff=unified&w=0#diff-ea35ea149a900d97cf2e45b19ead10214ce83e7d433453e3772e4437bd38a117R1-R28))
*  Create variable classes for each state-specific addition and subtraction to FAGI, with value type, entity, label, unit, definition period, reference, and state code ([link](https://github.com/PolicyEngine/policyengine-us/pull/3341/files?diff=unified&w=0#diff-f881add18867c6ed44cf304b16cc8c076bcee2d61148751fd8407b5f33282166R1-R14), [link](https://github.com/PolicyEngine/policyengine-us/pull/3341/files?diff=unified&w=0#diff-0fb3dc2cd1adcdb947f9000a86520a28ac0705a759e51152f0a8a764306db411R1-R15), [link](https://github.com/PolicyEngine/policyengine-us/pull/3341/files?diff=unified&w=0#diff-ab48dc8dcb6863062d89ed384affeec5a6de2014044e1dbf35239551249dc398R1-R15), [link](https://github.com/PolicyEngine/policyengine-us/pull/3341/files?diff=unified&w=0#diff-10c35ae132cb0b3e9d3d7a80ae87843ec48dff91cffff25f316293684d3d1a84R1-R15))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/subtractions.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+description: Virginia subtracts these sources from adjusted gross income.
+values:
+  2021-01-01:
+    - military_service_income
+    - va_military_basic_pay_subtraction
+    - 
+
+metadata:
+  unit: list
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/subtractions.yaml:9]:
```suggestion
  unit: list
  period: year
```

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/subtractions.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  label: West Virginia subtractions
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/subtractions.yaml:10]:
```suggestion
  label: West Virginia adjusted gross income subtractions
```

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/subtractions.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+description: Virginia subtracts these sources from adjusted gross income.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/subtractions.yaml:1]:
```suggestion
description: Virginia subtracts these sources from federal adjusted gross income.
```

File: policyengine_us/variables/gov/states/va/tax/income/subtractions/va_unemployement_compensation_benefit_subtraction.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/subtractions/va_unemployement_compensation_benefit_subtraction.py:1]: delete

File: policyengine_us/variables/gov/states/va/tax/income/subtractions/va_retirement_plan_taxed_subtraction.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/subtractions/va_retirement_plan_taxed_subtraction.py:1]: delete

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/subtractions.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    - va_military_benefit_subtraction # Certification Number 60
+
+
+metadata:
+  unit: list
+  period: year
+  label: West Virginia adjusted gross income subtractions
+  reference:
+    - title: 2022 Virginia Resident Form 760 Individual Income Tax Return
+      href: https://www.tax.virginia.gov/sites/default/files/taxforms/individual-income-tax/2022/760-2022.pdf#page=2
```
[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/va/tax/income/subtractions/subtractions.yaml:24]: https://www.tax.virginia.gov/sites/default/files/taxforms/individual-income-tax/2022/schedule-adj-2022.pdf

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/subtractions.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    - military_service_income # Special instructions for members of the military
+    - va_military_basic_pay_subtraction # Special instructions for members of the military
+    - va_disability_income_subtraction
+    - railroad_benefits # Certification Number 22
+    - military_service_income # Certification Number 28
+    - unemployment_compensation # Certification Number 37
+    - va_federal_state_employees_subtraction # Certification Number 39
+    - long_term_capital_gains # Certification Number 52
+    - student_loan_interest # Certification Number 55
+    - va_real_estate_investment_trust_subtraction # Certification Number 57
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/subtractions.yaml:14]:
```suggestion
```

File: policyengine_us/parameters/gov/states/va/tax/income/subtractions/subtractions.yaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    - va_military_benefit_subtraction # Certification Number 60
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/va/tax/income/subtractions/subtractions.yaml:15]:
```suggestion
    - va_military_benefit_subtraction # Certification Number 60
    - investment_in_529_plan # Line 34
```

File: policyengine_us/variables/gov/states/va/tax/income/va_agi.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    label = "Virginia Adjusted Gross Income (VAGI)"
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://law.lis.virginia.gov/vacodefull/title58.1/chapter3/article2/"
+    )
+    defined_for = StateCode.VA
+
+    adds = [
+        "adjusted_gross_income",
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/va/tax/income/va_agi.py:16]: include the va_age_deduction in the calculation 
include the taxable_social_security in the calc

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/va/tax/income/va_agi.py (hunk 7 above)
Comment (MaxGhenis) [reply policyengine_us/variables/gov/states/va/tax/income/va_agi.py:16]: i don't understand these. federal AGI already includes SS, and why would a deduction be an addition?

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/va/tax/income/va_agi.py]: flipped, should be subtractions

File: changelog_entry.yaml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+      - Virginia Adjusted Gross Income (VAGI).
```
Comment (MaxGhenis) [inline changelog_entry.yaml:4]:
```suggestion
      - Virginia Adjusted Gross Income.
```

File: policyengine_us/variables/gov/states/va/tax/income/va_agi.py [hunk 9]
```diff
@@ -8,5 +8,15 @@ class va_agi(Variable):
...
+        "https://law.lis.virginia.gov/vacodefull/title58.1/chapter3/article2/",
     )
+    defined_for = StateCode.VA
+
+    adds = ["adjusted_gross_income", "va_additions"]
+
+    subtracts = [
+        "va_subtractions",
+        "va_age_deduction",
+        "taxable_social_security",
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/va/tax/income/va_agi.py:21]: why not put in subtractions.yaml? it's separate on the form but included with other subtractions in the legal code 58.1-322.02(3)

File: policyengine_us/variables/gov/states/va/tax/income/subtractions/va_real_estate_investment_trust_subtraction.py
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/va/tax/income/subtractions/va_real_estate_investment_trust_subtraction.py:1]: for future reference we don't need variables like this. please check https://taxsim.nber.org for requirements

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**